lib/hipchat/room.rb

```diff
@@ -159,8 +159,8 @@ def send(from, message, options_or_notify = {})
     end
 
     def share_link(from, message, link)
-      if from.length > 15
-        raise UsernameTooLong, "Username #{from} is `#{from.length} characters long. Limit is 15'"
+      if from.length > 20
```

Comment:
This won't work, as the v2 API doesn't support a 'from' parameter when sharing a link (https://www.hipchat.com/docs/apiv2/method/share_link_with_room), and v1 has a limit of 15. Same for the send_file changes.